### PR TITLE
LogFile generation and log output improvment

### PR DESCRIPTION
**Pull request by Nocccer**
To avoid asking people to run heroiv from terminal, output are now written to a log file in .config/heroic/Logs.
Every month older logs are removed to prevent filling the Logs folder. Also added more LogPrefix's and adapt every log call to the new function syntax. Also added more error logging and minor fixes.

New Terminal Output:
```shell
INFO: [Legendary]: Location: /tmp/.mount_Heroic0wEAIF/resources/app.asar.unpacked/build/bin/linux/legendary
INFO: [Backend]: 
  Heroic Version: 2.0.2 Roronoa Zoro
  Legendary Version:  0.20.24 Follow Freeman!
  OS: Pop KERNEL: 5.15.11-76051511-generic ARCH: x64
  CPU: AMD Ryzen 7 3700X 8-Core Processor @3.6 GOVERNOR: ondemand
  RAM: Total: 16.8 GB Available: 11 GB
  GRAPHICS: GPU0: NVIDIA GeForce RTX 2060 SUPER VRAM: 7974MB DRIVER: 470.86 
  PROTOCOL: x11
  
WARNING: [Backend]: Protocol already registered.
[115472:0120/224121.876876:ERROR:browser_main_loop.cc(269)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
[115472:0120/224121.876953:ERROR:browser_main_loop.cc(269)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
[115472:0120/224121.877007:ERROR:browser_main_loop.cc(269)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
WARNING: [Backend]: python command not found
INFO: [Backend]: Command 'python3' found. Version: '3.8.10'
INFO: [Frontend]: Refreshing Library
INFO: [Legendary]: Refreshing library...
INFO: [Legendary]: Refreshing Epic Games...
INFO: [Legendary]: Refreshing Epic Games...
INFO: [Legendary]: [cli] INFO: Logging in...

INFO: [Legendary]: [Core] INFO: Trying to re-use existing login session...

INFO: [Legendary]: [cli] INFO: Getting game list... (this may take a while)

INFO: [Legendary]: Updating game list
INFO: [Legendary]: Game List Updated
INFO: [Legendary]: Checking for game updates
INFO: [Legendary]: Found 0 game(s) to update
INFO: [Backend]: checking for heroic updates
INFO: [Backend]: launching Control Calluna
WARNING: [Backend]: GameMode not installed
INFO: [Legendary]: Launch Command:   DXVK_HUD=fps           /tmp/.mount_Heroic0wEAIF/resources/app.asar.unpacked/build/bin/linux/legendary launch Calluna   --wine /usr/bin/wine --wine-prefix '/home/niklas/.wine'  
INFO: [Backend]: Log was written to /home/niklas/.config/heroic/GamesConfig/Calluna-lastPlay.log
INFO: [DXVKInstaller]: Removing dxvk version information
INFO: [DXVKInstaller]: dxvk removed from /home/niklas/.wine
INFO: [DXVKInstaller]: installing dxvk on... /home/niklas/.wine
INFO: [DXVKInstaller]: dxvk installed on /home/niklas/.wine
```

The related log file:
[heroic-2022-01-20T21:41:20.760Z.log](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/files/7908686/heroic-2022-01-20T21.41.20.760Z.log)

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [x] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [x] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

**Comment by dragonDScript** [comment]
Thanks. Will check the code later.

File: electron/main.ts [hunk 1]
```diff
@@ -1090,13 +1108,18 @@ ipcMain.handle('syncSaves', async (event, args) => {
...
+      LogPrefix.Backend
+    )
     return
   }
 
   const { stderr, stdout } = await Game.get(appName).syncSaves(arg, path)
-  logInfo(`${stdout}`)
-  logError(`${stderr}`)
+  logInfo(`${stdout}`, LogPrefix.Backend)
+  if (stderr.includes('ERROR')) {
```
Comment (Nocccer) [inline electron/main.ts:1120]:
```suggestion
  if (stderr.length) {
```

**Comment by flavioislima** [comment]
I will check this pr during the weekend. just one comment:
On several places that you see a check on:
`stderr.includes(ERROR) `its because most of legendary output, **ERROR or NOT**, comes from stderr, not stdout. So avoid removing those checks otherwise we can break the error handling.

**Comment by Nocccer** [comment]
I will have a lock over the legendary stdout and stderr logging and adapt it, if needed.

**Review by flavioislima — APPROVED** [review]
Looks nice!

**Pull request merged**